src/agent/onefuzz-telemetry/src/lib.rs

```diff
@@ -475,17 +475,17 @@ pub fn set_appinsights_clients(
     global::set_clients(instance_client, microsoft_client);
 }
 
-/// Try to submit any pending telemetry with a blocking call.
+/// Try to submit any pending telemetry.
 ///
 /// Meant for a final attempt at flushing pending items before an abnormal exit.
 /// After calling this function, any existing telemetry client will be dropped,
 /// and subsequent telemetry submission will be a silent no-op.
-pub fn try_flush_and_close() {
+pub async fn try_flush_and_close() {
     let clients = global::take_clients();
 
     for client in clients {
-        client.flush_channel();
-        client.close_channel();
+        // close_channel performs an implicit flush
+        client.close_channel().await;
     }
 }
```

Comment:
Maybe out of scope for this PR but now that this is async we could do this in parallel for some free perf!

```rust
tokio::join!(clients
        .iter()
        .map(|client| tokio::spawn(client.close_channel().await)));
```

Reply:
I don't think we gain a lot of perf with the move to async. This is meant to be a terminal call anyway. 
And it also introduce some weird asymetry. 
just like in the change in `src/agent/onefuzz-agent/src/main.rs`: the logging is creating without a runtime but we need a runtime to flush it. The non async version fit that scenario better. 
I propose we have both version async and non async and use them depending on if we are in an async context or not